datastore/mqtt/mqtt_test.go

```diff
@@ -0,0 +1,213 @@
+	return m.PublishFunc(topic, qos, retained, payload)
+}
+
+func (m *MockMQTTClient) Disconnect(quiesce uint) {
+	m.DisconnectFunc(quiesce)
+}
+
+func (m *MockMQTTClient) IsConnected() bool {
+	return m.IsConnectedFunc()
+}
+
+func (m *MockMQTTClient) IsConnectionOpen() bool {
+	return m.IsConnectionOpenFunc()
+}
+
+func (m *MockMQTTClient) Subscribe(topic string, qos byte, callback pahomqtt.MessageHandler) pahomqtt.Token {
+	return m.SubscribeFunc(topic, qos, callback)
+}
+
+func (m *MockMQTTClient) SubscribeMultiple(filters map[string]byte, callback pahomqtt.MessageHandler) pahomqtt.Token {
+	return m.SubscribeMultipleFunc(filters, callback)
+}
+
+func (m *MockMQTTClient) Unsubscribe(topics ...string) pahomqtt.Token {
+	return m.UnsubscribeFunc(topics...)
+}
+
+func (m *MockMQTTClient) AddRoute(topic string, callback pahomqtt.MessageHandler) {
+	m.AddRouteFunc(topic, callback)
+}
+
+func (m *MockMQTTClient) OptionsReader() pahomqtt.ClientOptionsReader {
+	return m.OptionsReaderFunc()
+}
+
+type MockToken struct {
+	WaitFunc        func() bool
+	WaitTimeoutFunc func(time.Duration) bool
+	DoneFunc        func() <-chan struct{}
+	ErrorFunc       func() error
+}
+
+func (m *MockToken) Wait() bool {
+	return m.WaitFunc()
+}
+
+func (m *MockToken) WaitTimeout(d time.Duration) bool {
+	return m.WaitTimeoutFunc(d)
+}
+
+func (m *MockToken) Done() <-chan struct{} {
+	return m.DoneFunc()
+}
+
+func (m *MockToken) Error() error {
+	return m.ErrorFunc()
+}
+
+var publishedTopics = make(map[string][]byte)
+
+func resetPublishedTopics() {
+	publishedTopics = make(map[string][]byte)
+}
+
+func mockPahoNewClient(o *pahomqtt.ClientOptions) pahomqtt.Client {
+	return &MockMQTTClient{
+
+		ConnectFunc: func() pahomqtt.Token {
+			return &MockToken{
+				WaitFunc:  func() bool { return true },
+				ErrorFunc: func() error { return nil },
+			}
+		},
+		IsConnectedFunc: func() bool {
+			return true
+		},
+		PublishFunc: func(topic string, qos byte, retained bool, payload interface{}) pahomqtt.Token {
+			publishedTopics[topic] = payload.([]byte)
+			return &MockToken{
+				WaitTimeoutFunc: func(d time.Duration) bool { return true },
+				WaitFunc:        func() bool { return true },
+				ErrorFunc:       func() error { return nil },
+			}
+		},
+	}
+}
+
+var _ = Describe("MQTTProducer", func() {
+	var (
+		mockLogger    *logrus.Logger
+		mockCollector metrics.MetricCollector
+		mockConfig    *mqtt.Config
+	)
+
+	BeforeEach(func() {
+		mockLogger, _ = logrus.NoOpLogger()
+		mockCollector = metrics.NewCollector(nil, mockLogger)
+		mockConfig = &mqtt.Config{
+			Broker:         "tcp://localhost:1883",
+			ClientID:       "test-client",
+			Username:       "testuser",
+			Password:       "testpass",
+			TopicBase:      "test/topic",
+			QoS:            1,
+			Retained:       false,
+			ConnectTimeout: 30,
+		}
+	})
+
+	Describe("Produce", func() {
+		It("should publish MQTT messages for each field in the payload", func() {
+			// Mock pahomqtt.NewClient
```

Comment:
dont think comments in this file are useful

Reply:
Ok, will remove them.